### PR TITLE
plat/arm: Fix misra warnings in platform code

### PR DESCRIPTION
**Pull request by satheesbalya-arm**
Change-Id: Ica944acc474a099219d50b041cfaeabd4f3d362f
Signed-off-by: Sathees Balya <sathees.balya@arm.com>

### CODE REVIEW

**Comment by dp-arm** [comment]
jenkins: test this please

File: plat/arm/board/fvp/fvp_topology.c [hunk 1]
```diff
@@ -16,15 +16,16 @@
...
+			assert_invalid_fvp_cluster_count);
 
 /*******************************************************************************
  * This function dynamically constructs the topology according to
  * FVP_CLUSTER_COUNT and returns it.
  ******************************************************************************/
 const unsigned char *plat_get_power_domain_tree_desc(void)
 {
-	unsigned int i;
+	int i;
```
Comment (soby-mathew) [inline plat/arm/board/fvp/fvp_topology.c]: why is the type changed here ?

File: plat/arm/common/arm_topology.c [hunk 2]
```diff
@@ -29,12 +29,14 @@ int arm_check_mpidr(u_register_t mpidr)
...
-	cpu_id = (mpidr >> MPIDR_AFF0_SHIFT) & MPIDR_AFFLVL_MASK;
+	cluster_id = (unsigned int) ((mpidr >> MPIDR_AFF1_SHIFT) &
+						MPIDR_AFFLVL_MASK);
+	cpu_id = (unsigned int) ((mpidr >> MPIDR_AFF0_SHIFT) &
+						MPIDR_AFFLVL_MASK);
 #endif /* ARM_PLAT_MT */
 
 	mpidr &= MPIDR_AFFINITY_MASK;
-	if (mpidr & valid_mask)
+	if ((mpidr & valid_mask) != 0ULL)
```
Comment (soby-mathew) [inline plat/arm/common/arm_topology.c]: 0U should be fine. Same comment for changes above as well.

**Comment by soby-mathew** [comment]
jenkins: test this please

File: plat/arm/board/fvp/fvp_topology.c (hunk 1 above)
Comment (satheesbalya-arm) [reply plat/arm/board/fvp/fvp_topology.c]: variable "i" is used in the for loop and compared with FVP_CLUSTER_COUNT 2 (signed). This gets rid of a MISRA warning.

plat/arm/board/fvp/fvp_topology.c
36 MISRA C-2012 Rule 10.4 (required) 251345
Essential type of the left hand operand "i" (unsigned) is not the same as that of the right operand "2U"(signed).

plat/arm/board/fvp/fvp_topology.c
37 MISRA C-2012 Rule 10.4 (required) 251345
Essential type of the left hand operand "i" (unsigned) is not the same as that of the right operand "2U"(signed).

**Pull request merged**